crates/store/re_sorbet/src/chunk_columns.rs

```diff
@@ -0,0 +1,165 @@
+    pub fn sanity_check(&self) {
+        for component in &self.components {
+            component.sanity_check();
+        }
+    }
+
+    /// Returns all indices and then all components;
+    /// skipping the `row_id` column.
+    ///
+    /// See also [`Self::get_index_or_component`].
+    // TODO(#9922): stop ignoring row_id
+    pub fn indices_and_components(&self) -> Vec<ColumnDescriptor> {
+        itertools::chain!(
+            self.indices.iter().cloned().map(ColumnDescriptor::Time),
+            self.components
+                .iter()
+                .cloned()
+                .map(ColumnDescriptor::Component),
+        )
+        .collect()
+    }
+
+    /// Index the index- and component columns, ignoring the `row_id` column completely.
+    ///
+    /// That is, `get_index_or_component(0)` will return the first index column (if any; otherwise
+    /// the first component column).
+    ///
+    /// See also [`Self::indices_and_components`].
+    // TODO(#9922): stop ignoring row_id
+    pub fn get_index_or_component(&self, index_ignoring_row_id: usize) -> Option<ColumnDescriptor> {
+        if index_ignoring_row_id < self.indices.len() {
+            Some(ColumnDescriptor::Time(
+                self.indices[index_ignoring_row_id].clone(),
+            ))
+        } else {
+            self.components
+                .get(index_ignoring_row_id - self.indices.len())
+                .cloned()
+                .map(ColumnDescriptor::Component)
+        }
+    }
+
+    /// Keep only the component columns that satisfy the given predicate.
+    #[must_use]
+    #[inline]
+    pub fn filter_components(mut self, keep: impl Fn(&ComponentColumnDescriptor) -> bool) -> Self {
+        self.components.retain(keep);
+        self
+    }
+}
+
+impl ChunkColumnDescriptors {
+    pub fn try_from_arrow_fields(
+        chunk_entity_path: Option<&EntityPath>,
+        fields: &ArrowFields,
+    ) -> Result<Self, SorbetError> {
+        Self::try_from(SorbetColumnDescriptors::try_from_arrow_fields(
+            chunk_entity_path,
+            fields,
+        )?)
+    }
+}
+
+impl TryFrom<SorbetColumnDescriptors> for ChunkColumnDescriptors {
+    type Error = SorbetError;
+
+    fn try_from(columns: SorbetColumnDescriptors) -> Result<Self, Self::Error> {
+        let SorbetColumnDescriptors { columns } = columns;
+
+        let mut row_ids = Vec::new();
+        let mut indices = Vec::new();
+        let mut components = Vec::new();
+
+        for column in &columns {
+            match column.clone() {
+                ColumnDescriptor::RowId(descr) => {
+                    if indices.is_empty() && components.is_empty() {
+                        row_ids.push(descr);
+                    } else {
+                        let err = format!(
+                            "RowId column must be the first column; but the columns were: {columns:?}"
+                        );
+                        return Err(SorbetError::custom(err));
```

Comment:
I see with have a pattern of using `SorbetError::Custom` for (mostly) chunk deserialisation error. After dealing with very bad error types (looking at you tonic), I'd rather not spread that any further.

Ideally, we'd have dedicated variant for all kinds of errors, but let's at least have a `SorbetError::ChunkDeserialisationError { reason: String }` variant and ideally get rid of `Custom`.